src/h5web/visualizations/line/config.ts

```diff
@@ -0,0 +1,26 @@
+import { Domain } from '../shared/models';
+import { StorageConfig, createPersistableState } from '../../storage-utils';
+
+interface LineVisConfig {
+  abscissaDomain: Domain | undefined;
+  setAbscissaDomain: (abscissaDomain: Domain | undefined) => void;
+  ordinateDomain: Domain | undefined;
+  setOrdinateDomain: (ordinateDomain: Domain | undefined) => void;
+}
+
+const STORAGE_CONFIG: StorageConfig = {
+  storageId: 'h5web:line',
+  itemsToPersist: [],
+};
+
+export const [useLineVisConfig] = createPersistableState<LineVisConfig>(
+  STORAGE_CONFIG,
+  set => ({
+    abscissaDomain: undefined,
+    setAbscissaDomain: (abscissaDomain: Domain | undefined) =>
+      set({ abscissaDomain }),
+    ordinateDomain: undefined,
+    setOrdinateDomain: (ordinateDomain: Domain | undefined) =>
+      set({ ordinateDomain }),
+  })
+);
```

Comment:
This was adapted from the `HeatmapConfig` to directly have to possiblity to use persistent storage. Even if it is not used yet...

Reply:
I don't get why you put the axis domains in there, though.

- The abscissa domain is definitely not a configuration option.
- The ordinate domain is like the data domain in the heatmap, I guess, but it's in the heatmap config store because of `customDomain`, and we don't have a configurable custom domain here. Seeing this, I'm actually reconsidering removing the data domain from the heatmap config and just using a memoised hook instead.

Reply:
Following your comment https://github.com/silx-kit/h5web/pull/117#discussion_r420615672, I will replicate the provider architecture